### ISSUE TITLE
Parameter Type breaks the package 

### ISSUE THREAD

**Issue by augustmuir** [mhrst/sign_in_with_apple_js#9]
Hey, thanks for the package. Super easy to use. The only thing is, I had to manually fix this error:

```
/C:/Users/Platypus/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/sign_in_with_apple_js-2.0.0/lib/sign_in_with_apple_js.dart:21:43: Error: The argument type 'Function' can't be assigned to the parameter type 'dynamic Function(Event)'.
 - 'Function' is from 'dart:core'.
 - 'Event' is from 'dart:html'.
    web_only.listenAppleIDSignInOnFailure(callback);
                                          ^

/C:/Users/Platypus/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/sign_in_with_apple_js-2.0.0/lib/sign_in_with_apple_js.dart:25:43: Error: The argument type 'Function' can't be assigned to the parameter type 'dynamic Function(Event)'.
 - 'Function' is from 'dart:core'.
 - 'Event' is from 'dart:html'.
    web_only.listenAppleIDSignInOnSuccess(callback);
```

I tried importing `dart:html` and `dart:core` and that didn't fix the issue. I left them imported (not sure if this is important), but what fixed it was removing the forced parameter type `Function`. It worked fine after doing that:

Fixed code:
```
StreamSubscription listenAppleIDSignInOnFailure(callback) => web_only.listenAppleIDSignInOnFailure(callback);

/// Listen to the success event from Sign in with Apple JS
StreamSubscription listenAppleIDSignInOnSuccess(callback) => web_only.listenAppleIDSignInOnSuccess(callback);
```

Code that doesn't work: 
```
/// Listen to the failure event from Sign in with Apple JS
StreamSubscription listenAppleIDSignInOnFailure(Function callback) =>
    web_only.listenAppleIDSignInOnFailure(callback);

/// Listen to the success event from Sign in with Apple JS
StreamSubscription listenAppleIDSignInOnSuccess(Function callback) =>
    web_only.listenAppleIDSignInOnSuccess(callback);
```

**Comment by mhrst** [comment]
Thanks for this!
I definitely see this problem with generically specifying `Function` as a type.  What also should work is:
```
listenAppleIDSignInOnFailure(void Function(dynamic) callback)
```

Testing this  and your fix for an update to release today.

**Comment by mhrst** [comment]
Resolved in e4d783bf7c763acb38edda37debc76c879ddbf60, version `2.0.1`

Thanks again for noticing this 👍🏼
